### PR TITLE
BRS - Encrypt Automation Account Variables

### PR DESCRIPTION
**Pull request by saurabhgautam-msft**

### CODE REVIEW

File: Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables [hunk 1]
```diff
@@ -0,0 +1,673 @@
...
+    Write-Host "[Step 4 of 5] Backing up Automation Account details"
+    Write-Host $([Constants]::SingleDashLine)
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {        
+        if(-not $SkipBackup)
+        {
+            # Backing up Automation Account details.
+            $backupFile = "$($backupFolderPath)\AutomationAccountsWithUnencryptedVariables.csv"
+            $automationAccountsWithUnEncryptedVariables | Export-CSV -Path $backupFile -NoTypeInformation
+            Write-Host "Automation Account details have been successful backed up to [$($backupFolderPath)]" -ForegroundColor $([Constants]::MessageType.Update)
```
Comment (jaiswalricha) [inline Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables:467]: can this be successfully backed up?

Comment (saurabhgautam-msft) [reply Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables:467]: Yeah, will update it in a sec.

File: Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables [hunk 2]
```diff
@@ -0,0 +1,673 @@
...
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    Write-Host "To encrypt the unencrypted variables associated with the Automation Accounts in a Subscription, Contributor or higher privileges on the Automation Accounts are required." -ForegroundColor $([Constants]::MessageType.Warning)
+    Write-Host $([Constants]::SingleDashLine)
+
+    Write-Host "[Step 2 of 5] Fetch all Automation Accounts"
+    Write-Host $([Constants]::SingleDashLine)
+    $automationAccountResources = @()
+    $requiredMinTLSVersion = 1.2
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables:250]: Is this needed?

Comment (saurabhgautam-msft) [reply Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables:250]: Yes, the step is needed.

I have addressed the two issue highlighted earlier i.e., now I am getting list of all automation accounts and verifying from that if an automation account is valid or not. And not saving variable information in local.

File: Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables [hunk 3]
```diff
@@ -0,0 +1,684 @@
...
+            return
+        } 
+
+        $allAutomationAccountResources = Get-AzAutomationAccount -ErrorAction SilentlyContinue
+        $validResources | ForEach-Object { 
+            try
+            {
+                $name = $_.ResourceName
+                $resourceGroupName = $_.ResourceGroupName
+                $automationAccountResources += $allAutomationAccountResources | Where-Object {$_.AutomationAccountName -eq $name -and $_.ResourceGroupName -eq $resourceGroupName}
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables:287]: If resource was there in file, but not present anymore in subscription. We should log that in skipped resource wth reason as resource not found.

File: Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables [hunk 4]
```diff
@@ -0,0 +1,684 @@
...
+            $automationAccountResourcesFromFile = Import-Csv -LiteralPath $FilePath
+            $validAutomationAccountResources = $automationAccountResourcesFromFile | Where-Object { ![String]::IsNullOrWhiteSpace($_.AutomationAccountName) -and ![String]::IsNullOrWhiteSpace($_.ResourceGroupName) }
+            $allAutomationAccountResources = Get-AzAutomationAccount -ErrorAction SilentlyContinue
+            $validAutomationAccountResources | ForEach-Object {
+                $resourceGroupName = $_.ResourceGroupName        
+                $name = $_.AutomationAccountName               
+
+                try
+                {
+                    $automationAccountResources += $allAutomationAccountResources | Where-Object {$_.AutomationAccountName -eq $name -and $_.ResourceGroupName -eq $resourceGroupName}
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables:333]: same here if no matching automation account let's log/print in console.

File: Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables (hunk 3 above)
Comment (saurabhgautam-msft) [reply Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables]: Addressed

File: Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables (hunk 4 above)
Comment (saurabhgautam-msft) [reply Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables]: Addressed

**Review by TarunKrShukla — APPROVED** [review]

**Review by jaiswalricha — APPROVED** [review]

**Pull request merged**